frontend/js/marc-record-merge-postmerge-service.js

```diff
@@ -447,6 +462,29 @@ export function removeExtra035aFromMerged(preferredRecord, otherRecord, mergedRe
   };
 }
 
+export function add080VersionCode (preferredRecord, otherRecord, mergedRecordParam) {
+  const mergedRecord = new MarcRecord(mergedRecordParam);
+
+  mergedRecord.fields.filter(field => field.tag === '080')
+    .forEach(field => {
+      const hasCode2 = field.subfields.findIndex(sub => sub.code === '2');
+      const hasCode9FKeep = field.subfields.findIndex(sub => sub.code === '9' && sub.value === 'FENNI<KEEP>');
+      if (hasCode2 < 0 && hasCode9FKeep > -1) {
+        let index = field.subfields.findIndex(sub => isNaN(sub.code) && sub.code >= 3);
```

Comment:
We want to find the index of the the numeric subfield that `2`-subfield will *preceed* and then pass that index to *splice* (Since the new element will be added to that index):
```js
const index = field.subfields.findIndex(({code}) => Number(code) >= 2)

if (index >= 0) {
  field.subfields.splice(index, 0, { code: '2', value: '1974/fin/fennica' });
} else {
  field.subfields.push({ code: '2', value: '1974/fin/fennica' });
}